### PR TITLE
MGMT-3611 Add proxy configuration in day2 clusters

### PR DESCRIPTION
**Pull request by empovit**

### CODE REVIEW

File: internal/cluster/proxy.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
+package cluster
```
Comment (masayag) [inline internal/cluster/proxy.go:1]: Nice refactor!

Comment (empovit) [reply internal/cluster/proxy.go:1]: :) pretty much just moving your code

**Review by masayag — COMMENTED** [review]
LGTM

File: tools/deploy_assisted_installer.py [hunk 2]
```diff
@@ -51,7 +51,9 @@ def main():
             data["spec"]["template"]["spec"]["containers"][0]["env"].append({'name':'OCM_TOKEN_URL', 'value': WIREMOCK_SERVICE + '/token'})
             data["spec"]["template"]["spec"]["containers"][0]["env"].append({'name':'OCM_SERVICE_CLIENT_ID', 'value': 'mock-ocm-client-id'})
             data["spec"]["template"]["spec"]["containers"][0]["env"].append({'name':'OCM_SERVICE_CLIENT_SECRET', 'value': 'mock-ocm-client-secret'})
-            data["spec"]["template"]["spec"]["containers"][0]["env"].append({'name':'ENABLE_KUBE_API', 'value': str(deploy_options.enable_kube_api).lower()})
+
+            if deploy_options.enable_kube_api:
```
[the remark this answers is not in the document]
Comment (empovit) [reply tools/deploy_assisted_installer.py:55]: No, just a small bug I encountered while running the subsystem

[the remark this answers is not in the document]
Comment (filanov) [reply tools/deploy_assisted_installer.py:55]: `enable_kube_api` have a default so why is it needed?

[the remark this answers is not in the document]
Comment (filanov) [reply tools/deploy_assisted_installer.py:55]: With this PR https://github.com/openshift/assisted-service/pull/1058 empty string will become false by default.

File: internal/cluster/proxy.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+// Diff detects if two proxy configurations differ.
+// Changed proxy settings mean that a new ISO file must be generated
+// to include the updated proxy settings
+func (p *Proxy) Diff(httpProxy, httpsProxy, noProxy string) bool {
+	if httpProxy != swag.StringValue(p.HTTPProxy) ||
+		httpsProxy != swag.StringValue(p.HTTPSProxy) ||
+		noProxy != swag.StringValue(p.NoProxy) {
+		return true
```
Comment (empovit) [inline internal/cluster/proxy.go:71]: This will return true even if no values where send via the REST API (i.e. no changes)

**Pull request closed without merging**